aiverify-foundation/moonshot-ui
# Feature/bookmark prompt

## Description

PR for `Save prompt as bookmark` feature.
A bookmark icon is added next to every prompt in red teaming session chat window. User can click on the icon and save the prompt as a `prompt bookmark` from a popup screen.

Code changes includes

- small refactor to  prompt talk bubble components
- add role attribute to Icon component for testing purposes
- rename a DialoguePairInfo type to PromptDetails 

## Motivation and Context

Useful feature - reuse prompts

## Type of Change


- feat: A new feature 










## How to Test

1. Start a new or Resume a Red Teaming Session
2. Next to the prompts in chat window, click on bookmark icon (ribbon icon)
3. Verify the details on the popup
4. Enter a name for the bookmark and save
5. Verify that the bookmark is added in the bookmarks sqlite db under moonshot data

## Checklist

Please check all the boxes that apply to this pull request using "x":

- [x ]  I have tested the changes locally and verified that they work as expected.
- [x ]  I have added appropriate unit tests or functional tests for the changes made.
- [x ]  I have followed the project's coding conventions and style guidelines.
- [x ]  I have rebased my branch onto the latest commit of the main branch.
- [x ]  I have added any necessary dependencies or packages to the project's build configuration.
- [x ]  I have performed a self-review of my own code.
- [x ]  I have read, understood and agree to the Developer Certificate of Origin below, which this project utilises.

## Screenshots (if applicable)

![image](https://github.com/user-attachments/assets/9395ec51-0543-4034-9e22-49ae47363671)

## Additional Notes

[Add any additional information or context that might be relevant to reviewers.]

<details>
  <summary>Developer Certificate of Origin</summary>

 ```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
 ```
</details>
